### DISCUSSION TITLE
[Q&A] 使用 API 中转服务，无法使用插件

### DISCUSSION THREAD

**Discussion by MartialBE** [lobehub/lobehub#928] [General Question] [answered]
### 💻 系统环境 | Operating System

macOS

### 🌐 浏览器 | Browser

Chrome

### 🐛 问题描述 | Bug Description

我将国内的`讯飞`/ `百度` 模型 转换成OpenAI API 格式，然后使用 `lobe` 调用插件时，`lobe` 无法正确识别函数调用。
好像是和匹配有关。 OpenAI 的 stream 会把 `tool_calls` 拆分了输出。 而国内的模型函数调用都是一次输出，所以我在转换的时候，函数调用也是一次性输出。

### 📷 复现步骤 | Recurrence Steps

使用的工具：https://github.com/MartialBE/one-api

开启 天气插件，询问天气：

下面是 我调用API输出的结果：

```
data: {"id":"as-fthf7qvbyg","object":"chat.completion.chunk","created":1704277500,"model":"ERNIE-Bot-4","choices":[{"index":0,"delta":{"tool_calls":[{"id":"as-fthf7qvbyg","type":"function","function":{"name":"realtime-weather____fetchCurrentWeather","arguments":"{\"city\":\"北京\"}"},"index":0}]},"finish_reason":"tool_calls"}]}

data: [DONE]


```

下面是 lobe 处理的输出结果：
```
{"tool_calls":[ {"id": "as-fthf7qvbyg", "type": "function", "function": {"name": "realtime-weather____fetchCurrentWeather", "arguments": "

```

lobe处理到 `arguments` 就没有下文了。导致无法正常调用插件。

<img width="1547" alt="iShot_2024-01-03_18 36 02" src="https://github.com/lobehub/lobe-chat/assets/42402987/68565b35-e5b1-48e9-8d51-ee3c2d264b3e">

<img width="1071" alt="iShot_2024-01-03_18 38 14" src="https://github.com/lobehub/lobe-chat/assets/42402987/7eb71fb7-ba39-43d2-af99-ae0eaec21bc5">

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
这个是不是应该提给 one-api ？我理解这个似乎是因为oneapi 没有按照openai标准方案输出fc 的调用吧？

[a comment by an automated account is not shown]

**Comment by MartialBE** [comment]
是我想错了。
我之前想的是 lobe 是否可以通过finish_reason=tool_calls来判断结束 tool_calls的拼装。所以应该可以支持一次性的tool_calls的读取。
这确实不是lobe的问题。我再去研究一下openai的stream输出。把oneapi那边输出的格式符合openai的规范。

[a comment by an automated account is not shown]
